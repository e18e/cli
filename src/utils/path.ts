export function normalizePath(path: string): string {
  return path.replaceAll('\\', '/');
}
