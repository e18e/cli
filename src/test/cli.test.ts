import {describe, it, expect, beforeAll, afterAll} from 'vitest';
import path from 'node:path';
import fs from 'node:fs/promises';
import {
  createTempDir,
  cleanupTempDir,
  createTestPackage,
  runCliProcess,
  stripVersion
} from './utils.js';
import {pack as packAsTarball} from '@publint/pack';

let mockTarballPath: string;
let tempDir: string;

beforeAll(async () => {
  // Create a temporary directory for the test package
  tempDir = await createTempDir();

  // Create a test package with some files
  await createTestPackage(tempDir, {
    name: 'mock-package',
    version: '1.0.0',
    type: 'module',
    main: 'index.js',
    dependencies: {
      'some-dep': '1.0.0'
    }
  });

  // Create a simple index.js file
  await fs.writeFile(
    path.join(tempDir, 'index.js'),
    'console.log("Hello, world!");'
  );

  // Create node_modules with a dependency
  const nodeModules = path.join(tempDir, 'node_modules');
  await fs.mkdir(nodeModules, {recursive: true});
  await fs.mkdir(path.join(nodeModules, 'some-dep'), {recursive: true});
  await fs.writeFile(
    path.join(nodeModules, 'some-dep', 'package.json'),
    JSON.stringify({
      name: 'some-dep',
      version: '1.0.0',
      type: 'module'
    })
  );

  // Pack the test package into a tarball (cross-platform, no external npm spawn)
  mockTarballPath = await packAsTarball(tempDir, {
    packageManager: 'npm',
    ignoreScripts: true,
    destination: tempDir
  });
});

afterAll(async () => {
  await cleanupTempDir(tempDir);
});

describe('CLI', () => {
  it('should run successfully with default options', async () => {
    const {stdout, stderr, code} = await runCliProcess(
      ['analyze', mockTarballPath],
      tempDir
    );
    if (code !== 0) {
      console.error('CLI Error:', stderr);
    }
    expect(code).toBe(0);
    expect(await stripVersion(stdout, process.cwd())).toMatchSnapshot();
    expect(stderr).toBe('');
  });

  it('should display package report', async () => {
    const {stdout, stderr, code} = await runCliProcess(
      ['analyze', mockTarballPath],
      tempDir
    );
    expect(code).toBe(0);
    expect(await stripVersion(stdout, process.cwd())).toMatchSnapshot();
    expect(stderr).toMatchSnapshot();
  });
});
