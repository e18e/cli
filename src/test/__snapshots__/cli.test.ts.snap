// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CLI > should display package report 1`] = `
"[36me18e[39m (cli <version>)

[90m┌[39m  Analyzing...
[90m│[39m
[34m●[39m  Summary
[90m│[39m  [36mPackage Name[39m        mock-package
[90m│[39m  [36mVersion[39m             1.0.0
[90m│[39m  [36mInstall Size[39m        170.0 B
[90m│[39m  [36mProd. Dependencies[39m  1
[90m│[39m  [36mDev. Dependencies[39m   0
[90m│[39m  [36mCJS Dependencies[39m    0
[90m│[39m  [36mESM Dependencies[39m    0
[90m│[39m
[34m●[39m  Results:
[90m│[39m
[90m│[39m  [34mSuggestions:[39m
[90m│[39m    [34m•[39m No type definitions found.
[90m│[39m
[90m│[39m
[90m└[39m  Done!

"
`;

exports[`CLI > should display package report 2`] = `""`;

exports[`CLI > should run successfully with default options 1`] = `
"[36me18e[39m (cli <version>)

[90m┌[39m  Analyzing...
[90m│[39m
[34m●[39m  Summary
[90m│[39m  [36mPackage Name[39m        mock-package
[90m│[39m  [36mVersion[39m             1.0.0
[90m│[39m  [36mInstall Size[39m        170.0 B
[90m│[39m  [36mProd. Dependencies[39m  1
[90m│[39m  [36mDev. Dependencies[39m   0
[90m│[39m  [36mCJS Dependencies[39m    0
[90m│[39m  [36mESM Dependencies[39m    0
[90m│[39m
[34m●[39m  Results:
[90m│[39m
[90m│[39m  [34mSuggestions:[39m
[90m│[39m    [34m•[39m No type definitions found.
[90m│[39m
[90m│[39m
[90m└[39m  Done!

"
`;
